daemon/graphdriver/overlay/overlay.go

```diff
@@ -248,6 +284,18 @@ func (d *Driver) Create(id, parent, mountLabel string, storageOpt map[string]str
 		}
 	}()
 
+	driver := &Driver{}
+	if err := d.parseStorageOpt(storageOpt, driver); err != nil {
+		return err
+	}
+
+	if driver.options.quota.Size > 0 && !strings.HasSuffix(dir, "-init") {
```

Comment:
@amir73il this code checks home directory doesn't have suffix "-init" 
(Create() function called for "-init" directory first and home directory next.